JenkinsfileStageMerge

```diff
@@ -1,34 +1,91 @@
-                    echo "No tag found for image ${it}!"
+pipeline {
+    agent {
+        kubernetes {
+            cloud 'openshift'
+            label 'merge-trigger-' + env.ghprbActualCommit
+            containerTemplate {
+                name 'jnlp'
+                args '${computer.jnlpmac} ${computer.name}'
+                image DOCKER_REPO_URL + '/' + openshiftProject + '/jenkins-continuous-infra-slave:' + STABLE_LABEL
+                ttyEnabled false
+                command ''
+            }
+        }
+    }
+    stages {
+        stage("Image Promotion") {
+            steps {
+                script {
+                    openshift.withCluster() {
+                        openshift.withProject(openshiftProject) {
+                            imageList.each {
+                                String tagList = sh(
+                                        script: "oc get is -n ${openshiftProject} -o=jsonpath=\'{.items[?(@.metadata.name==\"${it}\")].status.tags[*].tag}\'",
+                                        returnStdout: true
+                                ).trim()
+                                def prTag = tagList.tokenize(' ').find { it == "PR-" + env.ghprbPullId }
+                                if (prTag != null) {
+                                    echo "Found PR tag: " + prTag + " for image ${it}"
+                                    echo "Updating ${it}:stable tag to point to ${prTag}"
+                                    openshift.tag("${openshiftProject}/${it}:${prTag}",
+                                            "${openshiftProject}/${it}:stable")
+                                    imageOperations.add("${it}:stable now points to ${it}:${prTag}")
+                                } else {
+                                    echo "No tag found for image ${it}!"
+                                }
+                            }
+                        }
+                    }
+                    sendPRCommentforTags(imageOperations)
                 }
             }
         }
+        stage("Merge PR") {
+            steps {
+                script {
+                    // need this for ghprb plugin since it is really
+                    // a post build step and it assumes the build is complete.
+                    currentBuild.result = 'SUCCESS'
+                }
+                step([$class: 'GhprbPullRequestMerge', allowMergeWithoutTriggerPhrase: false, deleteOnMerge: false, disallowOwnCode: false, failOnNonMerge: false, mergeComment: 'testing the merge', onlyAdminsMerge: false])
+            }
+        }
     }
+}
 
-    step([$class: 'GhprbPullRequestMerge', allowMergeWithoutTriggerPhrase: false, deleteOnMerge: false, disallowOwnCode: false, failOnNonMerge: false, mergeComment: 'testing the merge', onlyAdminsMerge: false])
+/**
+ * Based on tagMap, add comment to GH with
+ * instructions to manual commands
+ *
+ * @param map of tags
+ * @return
+ */
+def sendPRCommentforTags(imageOperationsList) {
```

Comment:
Shouldn't this be added to the PipelineUtils or some new library instead of in the Jenkinsfile?